src/components/CardContainer/reducers/cardContainerReducer.ts

```diff
@@ -0,0 +1,25 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { fetchPopularMovies } from "../api/services/fetchPopularMovies";
+
+const cardContainerSlice: any = createSlice({
+  name: "cards",
+  initialState: {
+    cards: [],
+    status: null,
+  },
+  reducers: {},
+  extraReducers: {
+    [fetchPopularMovies.pending ]: (state: { status: string; }) => {
+      state.status = "loading";
+    },
+    [fetchPopularMovies.fulfilled]: (state: { status: string; cards: any; }, action: { payload: any; }) => {
```

Comment:
adding types like this is a bad idea. Crate interface for state and action

Reply:
done